services/ux-backend/handlers/featureflags/handler.go

```diff
@@ -0,0 +1,214 @@
+	}
+
+	type flagResult struct {
+		Value bool   `json:"value"`
+		Error string `json:"error,omitempty"`
+	}
+
+	result := make(map[string]flagResult)
+	for _, flagName := range flagNames {
+		switch flagName {
+		case flagNoobaa:
+			value, err := checkNoobaaFlag(r.Context(), client, namespace)
+			if err != nil {
+				klog.Errorf("failed to check noobaa flag: %v", err)
+				result[flagName] = flagResult{Value: false, Error: err.Error()}
+			} else {
+				result[flagName] = flagResult{Value: value}
+			}
+		case flagRGW:
+			value, err := checkRGWFlag(r.Context(), client, namespace)
+			if err != nil {
+				klog.Errorf("failed to check rgw flag: %v", err)
+				result[flagName] = flagResult{Value: false, Error: err.Error()}
+			} else {
+				result[flagName] = flagResult{Value: value}
+			}
+		default:
+			result[flagName] = flagResult{Value: false, Error: "unsupported flag"}
+		}
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+
+	allFlagsFailed := true
+	for _, res := range result {
+		if res.Error == "" {
+			allFlagsFailed = false
+			break
+		}
+	}
+
+	// if all flags failed, returning "500" as response
+	// if some flags succeeded, returning "200" with partial results and error details
+	if allFlagsFailed && len(result) > 0 {
+		w.WriteHeader(http.StatusInternalServerError)
+	} else {
+		w.WriteHeader(http.StatusOK)
+	}
+
+	if err := json.NewEncoder(w).Encode(result); err != nil {
+		klog.Errorf("failed to encode response: %v", err)
+		http.Error(w, "Internal server error", http.StatusInternalServerError)
+		return
+	}
+}
+
+func checkNoobaaFlag(ctx context.Context, client ctrlclient.Client, namespace string) (bool, error) {
+	noobaaExists, err := checkNoobaaCRExists(ctx, client, namespace)
+	if err != nil {
+		return false, fmt.Errorf("failed to check NooBaa CR: %w", err)
+	}
+	if !noobaaExists {
+		return false, nil
+	}
+
+	storageClassExists, err := checkNoobaaStorageClassExists(ctx, client, namespace)
+	if err != nil {
+		return false, fmt.Errorf("failed to check NooBaa StorageClass: %w", err)
+	}
+
+	return storageClassExists, nil
+}
+
+func checkNoobaaCRExists(ctx context.Context, client ctrlclient.Client, namespace string) (bool, error) {
+	klog.Info("Checking NooBaa CR existence in namespace:", namespace)
+
+	noobaaList := &nbv1alpha1.NooBaaList{}
+	if err := client.List(ctx, noobaaList, ctrlclient.InNamespace(namespace), ctrlclient.Limit(1)); err != nil {
+		return false, fmt.Errorf("failed to list NooBaa CRs: %w", err)
+	}
+
+	if len(noobaaList.Items) > 0 {
+		klog.Info("Found NooBaa CRs in namespace:", namespace, "count:", len(noobaaList.Items))
+		return true, nil
+	}
+
+	klog.Info("No NooBaa CRs found in namespace:", namespace)
+	return false, nil
+}
+
+func checkNoobaaStorageClassExists(ctx context.Context, client ctrlclient.Client, namespace string) (bool, error) {
+	klog.Info("Checking for NooBaa StorageClass existence")
+	
+	provisioner := fmt.Sprintf("%s.noobaa.io/obc", namespace)
```

Comment:
provisioner name won't depend on namespace it's always hardcoded https://github.com/red-hat-storage/ocs-operator/blob/main/controllers/util/csi.go#L4-L8

Reply:
Yes, I noticed this file.
But, I prefer not to hardcode, unless there is absolutely no other way out.

Reply:
Okay, I think there is one case where it can break. What happens in ROSA ?? It is still "openshift-storage", right ?? If so, then probably hardcoding is the only way out.

Reply:
Also, [csi.go#L4-L8](https://github.com/red-hat-storage/ocs-operator/blob/main/controllers/util/csi.go#L4-L8) is for Ceph, what about NooBaa, does that use "openshift-storage" prefix in ROSA as well ??

Reply:
Another thing, can these provisioner prefix be overridden by users via some spec or env variable or something similar, does that happen in downstream ??

Reply:
Or, I keep indexing and if a StorageClass is not found using the index, I fallback to list and filter, just to be sure. But, that is not a clean code, rather a mid-way between 2 approaches which is guaranteed to always work in case of edge cases.